src/Draco.LanguageServer/Capabilities/CodeCompletion.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CompletionService = Draco.Compiler.Api.CodeCompletion.CompletionService;
+using KeywordCompletionProvider = Draco.Compiler.Api.CodeCompletion.KeywordCompletionProvider;
+using ExpressionCompletionProvider = Draco.Compiler.Api.CodeCompletion.ExpressionCompletionProvider;
+using MemberAccessCompletionProvider = Draco.Compiler.Api.CodeCompletion.MemberAccessCompletionProvider;
+using Draco.Lsp.Model;
+using Draco.Lsp.Server.Language;
+
+namespace Draco.LanguageServer;
+
+internal sealed partial class DracoLanguageServer : ICodeCompletion
+{
+    public CompletionRegistrationOptions CompletionRegistrationOptions => new()
+    {
+        DocumentSelector = this.DocumentSelector,
+        TriggerCharacters = new[] { "." }
+    };
+
+    public Task<IList<CompletionItem>> CompleteAsync(CompletionParams param, CancellationToken cancellationToken)
+    {
+        var cursorPosition = Translator.ToCompiler(param.Position);
+        return Task.FromResult<IList<CompletionItem>>(this.completionService.GetCompletions(this.syntaxTree, this.semanticModel, cursorPosition).Select(Translator.ToLsp).ToList());
```

Comment:
Please factor out `this.completionService.GetCompletions(this.syntaxTree, this.semanticModel, cursorPosition)` into a local to make it easier to read